Sources/Amplitude/Amplitude.m

```diff
@@ -494,8 +495,7 @@ - (void) checkInForeground {
                     [self refreshDynamicConfig];
 
                     NSNumber *now = [NSNumber numberWithLongLong:[[self currentTime] timeIntervalSince1970] * 1000];
-                    [self startOrContinueSessionNSNumber:now];
-                    self->_inForeground = YES;
+                    [self startOrContinueSessionNSNumber:now inForeground:NO];
```

Comment:
Why we set _inForeground = YES, but pass inForeground:NO in the `startOrContinueSessionNSNumber` function?

Reply:
Before:
assumption that `_inForeground` is `NO` here
`startOrContinueSessionNSNumber` uses `_inForeground` value (`NO`) internally
set `_inForeground` to `YES` 

Possible issue: the assumption can be false if `enterBackground` wasn't completed successfully vs not executed at all for some reason and `_inForeground` is still `YES`

Now:
`startOrContinueSessionNSNumber` uses explicit value `NO` and do not depend on current `_inForeground` value